apps/courses/src/routes/users/dashboard/Mentor.tsx

```diff
@@ -352,27 +352,67 @@ export const MentorTabs = ({ courses, mentor }) => {
     );
   };
 
-  const MyActivity = () => {
-    const user: firebase.User = useUser();
-    const db = useFirestore();
-    const dbReviewsRef = db
+  const loadReviewData = async (db, uId, startAt, pageSize) => {
+    let dbRef = db
       .collection('users')
-      .doc(user.uid)
+      .doc(uId)
       .collection('reviews')
       // .where('status', '!=', 'pending')
       .orderBy('started_at', 'desc')
-      .limit(10);
-    const dbReviews: MentorReview[] = useFirestoreCollectionData(dbReviewsRef);
+      .limit(pageSize + 1);
+    if (startAt) {
+      dbRef = dbRef.startAt(startAt);
```

Comment:
Should we use `startAfter` instead of `startAt`?

Reply:
I've just read the [official doc](https://googleapis.dev/nodejs/firestore/latest/CollectionReference.html#startAt) on `startAt` and `startAfter`. The variable `startAt`, representing the last review, should be a doc snapshot.

Here, the object resulting from the `doc.data()` call is being passed to `startAt`. If the doc snapshot is used instead, pagination will work as intended.